File: bunny_rabbit.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature qw(:5.34);
use experimental qw(signatures);
use Algorithm::Combinatorics qw(permutations combinations);

# solve BUNNY + BUNNY = RABBIT
my @var = split '', 'BUNYRAIT';

my $citer = combinations([0..9], 8);
while (my $c = $citer->next) {
    my $piter = permutations(\@var);
    while (my $p = $piter->next) {
        my %h = map { $p->[$_] => $c->[$_] } 0..7;
        next unless $h{R} == 1;
        my $bunny = bunny(%h);
        my $rabbit = rabbit(%h);
        show_result($bunny, $rabbit) if is_result($bunny, $rabbit);
    }
}

sub bunny(%h) {
    return "$h{B}$h{U}$h{N}$h{N}$h{Y}";
}

sub rabbit(%h) {
    return "$h{R}$h{A}$h{B}$h{B}$h{I}$h{T}";
}

sub is_result($bunny, $rabbit) {
    return $bunny * 2 == $rabbit;
}

sub show_result($bunny, $rabbit) {
    say " $bunny";
    say "+$bunny";
    say $rabbit;
    say '';
}

File: bunny_rabbit.py
#!/usr/bin/env python
from itertools import permutations, combinations

def bunny(d):
    return (d['B'] * 10000 +
            d['U'] * 1000 +
            d['N'] * 100 +
            d['N'] * 10 +
            d['Y'])

def rabbit(d):
    return (d['R'] * 100000 +
            d['A'] * 10000 +
            d['B'] * 1000 +
            d['B'] * 100 +
            d['I'] * 10 +
            d['T'])

def is_result(b, r):
    return b * 2 == r

def show_result(b, r):
    print(f' {b}')
    print(f'+{b}')
    print(r)
    print()

variables = 'BUNYRAIT'
for c in combinations(range(10), 8):
    for p in permutations(variables):
        d = { z[0]: z[1] for z in zip(p, c) }
        if d['R'] != 1:
            continue
        b = bunny(d)
        r = rabbit(d)
        if is_result(b,r):
            show_result(b, r)
